app/autotranslate/server/autotranslate.js

```diff
@@ -1,29 +1,76 @@
 import { Meteor } from 'meteor/meteor';
-import { HTTP } from 'meteor/http';
 import _ from 'underscore';
 import s from 'underscore.string';
 
 import { settings } from '../../settings';
 import { callbacks } from '../../callbacks';
 import { Subscriptions, Messages } from '../../models';
 import { Markdown } from '../../markdown/server';
+import { Logger } from '../../logger';
+
+export class TranslationProviderRegistry {
+	static registerProvider(provider) {
+		// get provider information
+		const metadata = provider._getProviderMetadata();
+		if (!TranslationProviderRegistry._providers) {
+			TranslationProviderRegistry._providers = {};
+		}
+		TranslationProviderRegistry._providers[metadata.name] = provider;
+	}
+
+	static getActiveProvider() {
+		return TranslationProviderRegistry._providers[TranslationProviderRegistry._activeProvider];
+	}
+
+	static loadActiveServiceProvider() {
+		settings.get('AutoTranslate_ServiceProvider', (key, value) => {
+			TranslationProviderRegistry._activeProvider = value;
+			// TODO RocketChat.AutoTranslate = TranslationProviderRegistry._providers[TranslationProviderRegistry._activeProvider];
```

Comment:
@vickyokrm Any particular reason for this?

Reply:
Since we use Registry pattern to handle multiple service providers, with this change we update the registry the active provider to the service provider that is selected in settings.

Reply:
I mean any reason for this to be commented out.  Is it not needed?  If so can we remove it?

Reply:
It is not required and has been already removed.